tests/regression/fn/fn_vertical_indirection.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <cstddef>
+#include <gtest/gtest.h>
+
+#include <gridtools/fn/cartesian.hpp>
+#include <gridtools/fn/unstructured.hpp>
+
+#include <fn_select.hpp>
+#include <test_environment.hpp>
+
+namespace {
+    using namespace gridtools;
+    using namespace fn;
+    using namespace literals;
+
+    struct derivative_stencil {
+        auto operator()() const {
+            return [](auto const &field, auto const &offset) {
+                const auto shifted = shift(field, unstructured::dim::vertical{}, deref(offset));
+                return deref(field) - deref(shifted);
+            };
+        }
+    };
+
+    constexpr auto parabolic = []([[maybe_unused]] auto horizontal_idx, auto vertical_idx) -> float {
+        const float x = float(vertical_idx);
+        const float xp = x + 0.5f;
+        const float y = 0.5f * xp * xp;
+        return y;
```

Comment:
```suggestion
        const float xp = k + 0.5f;
        return 0.5f * xp * xp;
```

Reply:
Simplified it.

What should we do with the cast? Implicit conversion from `int` to `float` can cause loss of precision, so Clang and GCC issue a warning with `-Wconversion` and MSVC by default. As much as I hate this warning for this kind of silly numerical code, it's helpful when you have to mix signed, unsigned and different precision integers for whatever reason.

Reply:
Oh, than just re-add the cast, no prob.

Reply:
I changed it like:
```c++
        const float k_shifted = float(k) + 0.5f;
        return 0.5f * k_shifted * k_shifted;
```
'cause the naming was kinda bad in the first place.